src/adapter/src/coord/sequencer.rs

```diff
@@ -2656,8 +2656,23 @@ impl Coordinator {
                 arity,
             },
         );
-        self.ship_dataflow(dataflow, cluster_id).await;
 
+        match self.ship_dataflow_fallible(dataflow, cluster_id).await {
+            Ok(_) => {}
+            Err(e) => {
+                self.active_conns
+                    .get_mut(&session.conn_id())
+                    .expect("must exist for active sessions")
+                    .drop_sinks
+                    .retain(|sink| sink.global_id != sink_id);
+                self.metrics
+                    .active_subscribes
+                    .with_label_values(&[session_type])
+                    .dec();
+                self.pending_subscribes.remove(&sink_id);
```

Comment:
This stuff will bitrot real fast. Why do we need to call these methods? If there's some common cleanup that needs to happen sometimes, can that get a dedicated function so all places do it correctly?

Reply:
All of these are populated directly above this method call. I just pushed a change to abstract away everything done here. Please take a look at the most recent commit.